datalad/support/annexrepo.py

```diff
@@ -506,7 +506,8 @@ def get_corresponding_branch(self, branch=None):
         else:
             return branch
 
-    def get_tracking_branch(self, branch=None, corresponding=True):
+    def get_tracking_branch(self, branch=None, remote_only=False,
+                            corresponding=True):
```

Comment:
Maybe swap order, such that `corresponding` remains the 2nd? I am torn whether backward-friendly keyword-less access is superior to GitRepo homogeneity or vice versa...

Reply:
Right, that's the main trade-off.  I originally went the other way, but then decided it'd be better to keep the parent-child API between GitRepo/AnnexRepo in agreement.  Beyond a general preference, I think it particularly makes sense in this case because otherwise we introduce a hard-to-detect case where `ds.repo.get_tracking_branch(x, y)` does different things depending on whether you're in a plain repo or an annex repo.  I didn't spot any actual callers that use `corresponding` (positional or keyword) in the main code base or any of our extensions.

Reply:
An alternative that results in a clearer failure mode:  We could make everything after `branch` keyword-only.

Reply:
After another night of sleep, I think your solution is the right one.